src/Catrobat/Controller/Web/ProgramController.php

```diff
@@ -543,6 +547,62 @@ private function createProgramDetailsArray(Program $program,
     ];
   }
 
+  /**
+   * @Route("/translate/project/{id}", name="translate_project", methods={"GET"})
+   *
+   * @throws \Doctrine\ORM\NoResultException
+   */
+  public function translateProjectAction(Request $request, string $id) {
+    if (!$request->query->has('target_language')) {
+      return new Response('Target language is required', Response::HTTP_BAD_REQUEST);
+    }
+
+    $project = $this->program_manager->find($id);
+
+    if (!$this->program_manager->isProjectVisibleForCurrentUser($project)) {
+      return new Response('No project found for this id', Response::HTTP_NOT_FOUND);
+    }
+
+    $source_language = $request->query->get('source_language');
+    $target_language = $request->query->get('target_language');
+
+    if ($source_language === $target_language) {
+      return new Response('Source and target languages are the same', Response::HTTP_UNPROCESSABLE_ENTITY);
+    }
+
+    $to_translate = [$project->getName(), $project->getDescription(), $project->getCredits()];
+    $translation_result = [];
+
+    foreach ($to_translate as $text) {
+      if (null == $text) {
+        array_push($translation_result, null);
+        continue;
+      }
+
+      try {
+        $translated_text = $this->translation_delegate->translate($text, $source_language, $target_language);
+      } catch(InvalidArgumentException $exception) {
+        return new Response($exception->getMessage(), Response::HTTP_BAD_REQUEST);
+      }
+
+      if (null === $translated_text) {
+        return new Response('Translation unavailable', Response::HTTP_SERVICE_UNAVAILABLE);
+      }
+
+      array_push($translation_result, $translated_text);
+    }
+
+    return new JsonResponse([
+      'id' => $project->getId(),
+      'source_language' => $source_language ?? $translation_result[0]->detected_source_language,
+      'target_language' => $target_language,
+      'translated_title' => $translation_result[0]->translation,
```

Comment:
Will the title ever be null? Like, will the $project->getName() ever return null?

Reply:
no. the database entry is not nullable